modules/genai-ecosystem/pages/langchain4j.adoc

```diff
@@ -40,6 +40,258 @@ An example is avalable at: https://github.com/langchain4j/langchain4j-examples/t
 include::https://github.com/langchain4j/langchain4j-examples/raw/main/neo4j-example/src/main/java/Neo4jEmbeddingStoreExample.java[]
 ----
 
+To create a Neo4jEmbeddingStore, we can write:
+[source,java]
+----
+Neo4jEmbeddingStore embeddingStore = Neo4jEmbeddingStore.builder().<builderParameters>.build();
+
+
+
+// we can use with `withBasicAuth`
+.withBasicAuth(neo4jContainer.getBoltUrl(), USERNAME, ADMIN_PASSWORD)
+// as alternative to driver:
+Driver driver = GraphDatabase.driver(neo4jContainer.getBoltUrl(), AuthTokens.basic(USERNAME, ADMIN_PASSWORD));
+----
+
+Where `<builderParameters>` must have `dimension` and either `driver` or `withBasicAuth` parameters,
+besides other optional ones.
+
+Here is the complete builder list:
+
+[options="header",cols="m,m,a"]
+|===
+| Key  | Default Value | Description
+| driver | <required if withBasicAuth is not set> | the https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/Driver.html[Java Driver instance]
+| withBasicAuth | <required if withBasicAuth is not set> | Creates an instance of Neo4jEmbeddingStore defining a https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/Driver.html[Java Driver instance], starting from `uri`, `user` and `password`
+| dimension | <required> | the vector's dimension
+| config | org.neo4j.driver.SessionConfig.forDatabase(`<databaseName>`) | the https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/SessionConfig.html[SessionConfig instance]
+| label | "Document" | the label name 
+| embeddingProperty | "embedding" | the embedding property name 
+| idProperty | "id" | the id property name
+| metadataPrefix | "" | the metadata prefix 
+| textProperty | "text" | the text property name
+| indexName | "vector" | the vector index name
+| databaseName | "neo4j" | the database name
+| retrievalQuery | "RETURN properties(node) AS metadata, node.`idProperty` AS `idProperty`, node.`textProperty` AS `textProperty`, node.`embeddingProperty` AS `embeddingProperty`, score" | the retrieval query
```

Comment:
retrieval query doesn't render correctly perhaps you need to wrap it in a code block ? or remove the individual backticks